OpenRA.Mods.Common/Widgets/Logic/Editor/ActorEditLogic.cs

```diff
@@ -234,6 +234,25 @@ public override void Tick()
 					var ownerHandler = new EditorActorOptionActionHandle<PlayerReference>(updateOwner, actor.Owner);
 					editActorPreview.Add(ownerHandler);
 
+					var specificOwnerInfo = actor.Info.TraitInfoOrDefault<RequiresSpecificOwnersInfo>();
+					Dictionary<string, PlayerReference>.ValueCollection owners;
+
+					if (specificOwnerInfo != null)
+					{
+						owners = editorActorLayer.Players.Players.Values.Where(x =>
+								specificOwnerInfo.ValidOwnerNames.Any(y => x.Name == y))
+							.ToDictionary(player => player.Name)
+							.Values;
+						if (owners.Count == 1)
+						{
+							ownerDropdown.IsDisabled = () => true;
+						}
```

Comment:
Single curly braces must be omitted according to https://github.com/OpenRA/OpenRA/wiki/Coding-Standard